pkg/model/requests_processing_test.go

```diff
@@ -68,24 +69,71 @@ func testRequestsProcessing(t *testing.T, describe spec.G, it spec.S) {
+		var request RequestEntity
 
-		it.Before(func() {
-			bufferStock := NewRequestsRoutingStock(envFake, NewReplicasActiveStock(), nil)
-			request = NewRequestEntity(envFake, bufferStock, RequestConfig{CPUTimeMillis: 200, IOTimeMillis: 200, Timeout: 3 * time.Second})
-			subject.Add(request)
+		describe("there is no free cpu resource", func() {
+			it.Before(func() {
+				*rawSubject.occupiedCPUCapacityMillisPerSecond = *rawSubject.totalCPUCapacityMillisPerSecond
+				routingStock := NewRequestsRoutingStock(envFake, NewReplicasActiveStock(), nil)
+				request = NewRequestEntity(envFake, routingStock, RequestConfig{CPUTimeMillis: 200, IOTimeMillis: 200, Timeout: 3 * time.Second})
+				subject.Add(request)
+			})
+			describe("scheduling processing", func() {
+				it("schedules a movement from RequestsProcessing to RequestsFailed", func() {
+					assert.Equal(t, simulator.StockName("RequestsFailed"), envFake.Movements[0].To().Name())
+				})
+			})
+
+			it("there is 1 entity in processingStock", func() {
+				assert.Equal(t, subject.Count(), uint64(1))
+			})
 		})
+		describe("request fails as request total time exceeds request timeout", func() {
+			it.Before(func() {
+				routingStock := NewRequestsRoutingStock(envFake, NewReplicasActiveStock(), nil)
+				request = NewRequestEntity(envFake, routingStock, RequestConfig{CPUTimeMillis: 20000, IOTimeMillis: 200, Timeout: 3 * time.Second})
+				subject.Add(request)
+			})
+
+			it("increments the number of requests since last Stat()", func() {
+				assert.Equal(t, int32(1), rawSubject.numRequestsSinceLast)
+			})
 
-		it("increments the number of requests since last Stat()", func() {
-			assert.Equal(t, int32(1), rawSubject.numRequestsSinceLast)
+			it("cpu resource is allocated", func() {
+				assert.NotEqual(t, rawSubject.occupiedCPUCapacityMillisPerSecond, 0)
+			})
+			it("schedules a movement from RequestsProcessing to RequestsFailed", func() {
+				assert.Equal(t, simulator.StockName("RequestsFailed"), envFake.Movements[0].To().Name())
+			})
+
+			describe("when request timeout is over", func() {
+				it("time for movement from RequestsProcessing to RequestsFailed = current time + timeout ", func() {
+					assert.Equal(t, envFake.TheTime.Add(request.(*requestEntity).requestConfig.Timeout), envFake.Movements[0].OccursAt())
+				})
+			})
 		})
 
-		describe("scheduling processing", func() {
+		describe("request completes as request total time doesn't exceed request timeout", func() {
+			it.Before(func() {
+				//there is free cpu resource
+				*rawSubject.occupiedCPUCapacityMillisPerSecond = 0.0
+				routingStock := NewRequestsRoutingStock(envFake, NewReplicasActiveStock(), nil)
+				request = NewRequestEntity(envFake, routingStock, RequestConfig{CPUTimeMillis: 200, IOTimeMillis: 200, Timeout: 3 * time.Second})
+				subject.Add(request)
+			})
+
+			it("cpu resource is allocated", func() {
+				assert.NotEqual(t, rawSubject.occupiedCPUCapacityMillisPerSecond, 0)
```

Comment:
Should we assert how much CPU was allocated?

Reply:
Fixed it. See commit 2d6552efb24f1c1eed205483f07717a4798bf296.